synapse/util/async_helpers.py

```diff
@@ -86,11 +86,8 @@ def errback(f):
 
         deferred.addCallbacks(callback, errback)
 
-    def observe(self):
+    def observe(self) -> defer.Deferred:
         """Observe the underlying deferred.
-
-        Can return either a deferred if the underlying deferred is still pending
-        (or has failed), or the actual value. Callers may need to use maybeDeferred.
```

Comment:
Can you doc what the function returns, even if it's less ambiguous now?

Reply:
Hmm, I guess it is a little unclear, though the actual return type is documented in the type hint. Let me add a comment.